src/toggle-switch/toggle-switch.scss

```diff
@@ -2,194 +2,206 @@
 // See LICENSE.md in the project root for license terms and full copyright notice.
 @import '../style/index';
 
+$toggle-switch-height: $iui-baseline * 2;
+
 @mixin iui-toggle-switch {
   @include iui-reset;
-  display: flex;
+  display: grid;
+  grid-template: 'toggle';
+  grid-column-gap: $iui-s;
   align-items: center;
   font-size: $iui-font-size;
   width: fit-content;
-  padding: ($iui-baseline/2) 0;
+  padding: ($iui-baseline * 0.5) 0;
   user-select: none;
   cursor: pointer;
   position: relative;
+  isolation: isolate;
   @include themed {
     color: rgba(t(iui-color-foreground-body-rgb), t(iui-opacity-2));
   }
 
-  > input {
-    width: 0;
-    height: 0;
+  > input[type='checkbox'] {
+    @include iui-reset;
+    @include iui-focus($thickness: 1px, $offset: 1px);
+    grid-area: toggle;
+    display: inline-block;
+    flex-shrink: 0;
+    position: relative;
     appearance: none;
-    opacity: 0;
-    position: absolute;
+    width: $toggle-switch-height * 2 - $iui-xs;
+    height: $toggle-switch-height;
+    border-radius: $toggle-switch-height * 0.5;
+    cursor: pointer;
+    @include themed {
+      background-color: t(iui-color-background-1);
+      border-color: rgba(t(iui-color-foreground-body-rgb), t(iui-opacity-4));
+    }
 
-    // Unchecked toggle switch
-    ~ .iui-toggle {
-      display: block;
-      position: relative;
-      height: $iui-baseline * 2 - $iui-xxs;
-      width: ($iui-m * 2) + ($iui-xxs * 3);
-      border-radius: $iui-baseline;
+    &::before {
+      content: '';
+      position: absolute;
+      inset: 0;
+      transition: border-color $iui-speed-fast ease-out;
+      border-radius: inherit;
+      border-style: solid;
+      border-width: 1px;
+      border-color: inherit;
+      box-sizing: border-box;
+    }
+
+    &::after {
+      content: '';
+      position: absolute;
+      height: $toggle-switch-height - ($iui-xxs * 3);
+      width: $toggle-switch-height - ($iui-xxs * 3);
+      aspect-ratio: 1 / 1;
+      top: 50%;
+      transform: translateY(-50%);
```

Comment:
I think this center could be improved by just using flex.

```scss
input {
  display: flex;
  ...
  &::after {
    align-self: center;
  }
}
```

Reply:
I see it's not even centered currently (on my 1440p screen)
![image](https://user-images.githubusercontent.com/9084735/152399108-c4533dfc-37c4-47ea-95a2-290cc36af366.png)

Reply:
I made some changes, can you check again?

Reply:
Still looks the same
![image](https://user-images.githubusercontent.com/9084735/152584878-3b3f347e-1f4e-40ab-8add-e31fe700c7c8.png)

Reply:
🤔 I'm at a loss here.

Reply:
Can you confirm if this is happening to all of the toggles on the demo page?  Or only the toggles within `.iui-input-group`?

Reply:
Looks like it's on both
![image](https://user-images.githubusercontent.com/9084735/153675601-db9ea7c7-e1c0-40a3-a593-c41238f4e500.png)
![image](https://user-images.githubusercontent.com/9084735/153675639-1e8f7082-11ce-42ad-8919-23b9a4a539e0.png)

My guess is that the wrapping `<label>` is causing the problem because it has a 22px height

Reply:
Pretty sure we discovered the cause of the off centering: `$iui-baseline` equaling 11px in combination with OS scaling.  Our toggle switch is `$iui-baseline * 2` in height, meaning 22px.  If a user's OS is set to scale at 125% then 22px * 1.25 scale = 27.5px tall.  It's that half pixel that is causing issues.  I am investigating changing our baseline from 11px to 12px but this will effect nearly all of our UI components.

Reply:
Size has been changed to 24px from 22px height, @mayank99 can you confirm if this is displaying centered when OS scaling is set to 125%?